app/scripts/modules/core/src/search/searchResult/AbstractBaseResultRenderer.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+import { capitalize } from 'lodash';
+
+import { NgReact } from 'core/reactShims';
+import { IIndexed } from 'core/domain/IIndexed';
+import { IResultRenderer } from './searchResultFormatter.registry';
+
+export interface ITableColumnConfigEntry<T> {
+  key: string;
+  label?: string;
+  defaultValue?: string;
+  notSortable?: boolean;
+  scopeField?: boolean;
+  cellRenderer: (item: T, key?: string, defaultValue?: string) => JSX.Element;
+}
+
+@autoBindMethods
+export abstract class AbstractBaseResultRenderer<T extends IIndexed> implements IResultRenderer {
+
+  private gridElement: HTMLElement;
+
+  public abstract getRendererClass(): string;
+  public abstract getKey(item: T): string;
+  public abstract getColumnConfig(): ITableColumnConfigEntry<T>[]
+
+  public sortItems(items: T[]): T[] {
+    return items;
+  }
+
+  public defaultCellRender(item: T, key: string): JSX.Element {
+    return (<div className={`${this.getRendererClass()}-${key}`} key={key}>{item[key]}</div>);
+  };
+
+  public hrefCellRenderer(item: T, key: string): JSX.Element {
+    return (
+      <div
+        key={key}
+        className={`${this.getRendererClass()}-${key}`}
+      >
+        <a href={item['href']}>{item[key]}</a>
+      </div>
+    );
+  }
+
+  public accountCellRenderer(item: T, key: string): JSX.Element {
+
+    let result: JSX.Element;
+    if (item[key]) {
+      const { AccountTag } = NgReact;
+      const accounts = item[key].split(',').sort().map((account: string) => (
+        <AccountTag key={account} account={account}/>));
+      result = (
+        <div className={`${this.getRendererClass()}-account`} key="env">
+          {accounts}
+        </div>
+      );
+    } else {
+      result = (<div key="unknown" className={`${this.getRendererClass()}-account`}>-</div>);
+    }
+
+    return result;
+  };
+
+  public valueOrDefaultCellRenderer(item: T, key: string, defaultValue = ''): JSX.Element {
+    return (<div className={`${this.getRendererClass()}-${key}`} key={key}>{item[key] || defaultValue}</div>);
```

Comment:
Single lines of JSX don't need the parentheses. I thought lint would break on it, but apparently not. I don't care either way, I should probably just delete this comment.

Reply:
I personally don't mind wrapping something like this in parens, and would be very angry with lint if it broke on this.

Reply:
i'm beginning to think you're the one who hates freedom.